Source/WebCore/rendering/svg/RenderSVGBlock.cpp

```diff
@@ -40,6 +40,19 @@ RenderSVGBlock::RenderSVGBlock(SVGGraphicsElement& element, RenderStyle&& style)
 {
 }
 
+void RenderSVGBlock::updateFromElement()
+{
+    RenderLayerModelObject::updateFromElement();
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    if (!document().settings().layerBasedSVGEngineEnabled() || !parent())
+        return;
+
+    updateFromStyle();
+    updateLayerTransform();
```

Comment:
Doesn't this mean updateFromStyle and updateLayerTransform get called twice for LBSE?

Reply:
Indeed - I could have mentioned this. The first call is nonsense, too early. styleDidChange() gets invoked just after the renderer received a style -- too early to compute SVG transform matrices. Due to the transform-box/transform-origin dependency the transform matrix of a container can only be fully known after children finished layout (and thus the container boundaries are well defined). However I agree, and it's good that you mention it, this should be addresses as well in this PR. I'll sleep about it :-)

Reply:
@rwlbuis you send me down the rabbit hole -- no more updateFromElement() calls, guaranteed one updateFromStyle() call per cycle, not multiple ones, only from one place, etc... much safer handling of the SVG transform flags, and most importantly: consistent for all LBSE renderers.

Reply:
I like this one much better :)